### PR TITLE
Xanderjakeq

### PR DESCRIPTION
**Pull request by xanderjakeq**

### CODE REVIEW

File: todoapp/src/App.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      })
+    this.props.addTodo(this.state.newtodo)
+  }
+
+  render() {
+    return (
+      <div className="App">
+        <div>
+            <form onSubmit = {this.handleSubmit}>
+              <input placeholder = 'New Task' type='text' name='newtodo' value = {this.state.newtodo} onChange = {this.handleChange}/>
```
Comment (sabrinafilipelli) [inline todoapp/src/App.js:58]: The toggle works, but the toggle box on the left side is not functional unless you are in the deleted list.

File: todoapp/src/App.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+              }) : this.props.todos.filter(todo => !todo.isCompleted).map((todo, idx) => {
+                return <Todo key = {todo.id} data = {todo} idx = {idx} toggleComplete = {this.props.toggleComplete} delete = {this.props.deleteTodo}/>
+              })}
+            </ul>
+        </div>
+
+        <div className = 'sideBar'>
+          <button onClick = {this.handleSubmit} className = 'addButton'>add</button>
+          <button onClick = {this.handleToggleAll} className = 'toggleDisplay'> {this.state.showAll ? <ToggleRight size = {30}/> : <ToggleLeft size = {30}/>} </button>
+          <ul className = 'stats'>
```
Comment (sabrinafilipelli) [inline todoapp/src/App.js:72]: These numbers need to have labels (i.e. Total, Completed, etc).

File: todoapp/src/App.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            <li><p>{this.props.todos.filter(todo => !todo.isCompleted).length}</p></li>
+            <li><p>{this.props.todos.filter(todo => todo.isCompleted).length}</p></li>
+            <li><p>{this.props.todos.length}</p></li>
```
Comment (sabrinafilipelli) [inline todoapp/src/App.js:75]: Maybe add a little bit of margin to the bottom of your list.

File: todoapp/src/App.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    }
+    this.setState({
+        newtodo: ''
+      })
+    this.props.addTodo(this.state.newtodo)
+  }
+
+  render() {
+    return (
+      <div className="App">
```
Comment (sabrinafilipelli) [inline todoapp/src/App.js:55]: Add a header or title? Label the toggle button?

**Review by sabrinafilipelli — APPROVED** [review]
### State Tree

- [x] Application should have an input field, a submit button, and a list of items that represents your todo list.
- [x] Application's state tree should have a single property called todos.
- [x] Created your store in src/index.js. The <Provider > component wraps <App /> and passes the created store into <Provider > as one of its properties. 

### React

- [x] Entering a new to-do should call an action creator that adds a new todo item to the todos array on the application state tree.
- [x] When the user presses submit it will invoke the appropriate action creator which will then have its new action fed through all of the reducers.
- [x] To-do list is displayed by creating a container that receives the application's todos array as a prop. That container then uses map to display the array.
- [x] When a to-do item is clicked an action will be dispatched that will toggle that to-do item's completed property to either true or false. 

### Stretch 

- [x] Implement the ability to delete todo list items. 
- [ ] Use localStorage to make the data persist.

Notes: Great job, Xander! Flawless execution of Redux.  A few minor issues pertaining to user experience, but nothing major.  Overall, excellent work!

⭐️ ⭐️ ⭐️

**Comment by sabrinafilipelli** [comment]
Thank you for deploying your application!

File: todoapp/src/App.js (hunk 1 above)
Comment (xanderjakeq) [reply todoapp/src/App.js:58]: By default, it shows incomplete tasks, the big toggle is for showing all tasks. So if unchecked, the task would show up again when toggled back to view only incompletes.

File: todoapp/src/App.js (hunk 2 above)
Comment (xanderjakeq) [reply todoapp/src/App.js:72]: Clutter xD

I think it self explanatory when being used.

File: todoapp/src/App.js (hunk 3 above)
Comment (xanderjakeq) [reply todoapp/src/App.js:75]: Yeah xD

**Pull request merged**